### PR TITLE
chore: ci deployment configuration

### PR DESCRIPTION
**Pull request by alanshaw**
Adds configuration for deploying to AWS.

* [x] Created `deploy.yml` GH action using shared-workflows
* [x] Updated `build.yml` GH action to use shared-workflows
* [x] Created `deploy` script to package.json
* [x] Created Github environments for prod and staging (with approval requirement for prod)
* [x] Created IAM roles for github actions
* [x] Created IAM roles & policies for staging/prod lambda functions
* [x] Added github repository secrets
* [x] Created prod/staging lambda and attached relevant roles
* [x] Added Dockerfile & metrics.yml

Did I miss anything?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
   "type": "module",
   "scripts": {
     "test": "ava --verbose",
-    "lint": "standard"
+    "lint": "standard",
+    "deploy": "aws lambda update-function-code --cli-connect-timeout 15000 --region $AWS_REGION --function-name $LAMBDA --image $AWS_ACCOUNT_ID.dkr.ecr.$AWS_REGION.amazonaws.com/$AWS_ECR_REPOSITORY:$DOCKER_IMAGE_VERSION"
```
Comment (alanshaw) [inline package.json:10]: yes? https://github.com/ipfs-elastic-provider/shared-workflows/issues/1

File: metrics.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+component: side-indexer-lambda
+interval: 1000
+metrics:
+  s3-reads: AWS S3 GetObject
+  s3-writes: AWS S3 PutObject
```
Comment (alanshaw) [inline metrics.yml:6]: I'm not 100% know what this is for or if it works...

Comment (francardoso93) [reply metrics.yml:6]: This demands some extra logic with OpenTelemetry for working. If you want to keep it simple (w/o metrics) remove it here and also from the Dockerfile Copy command

File: package.json (hunk 1 above)
Comment (francardoso93) [reply package.json:10]: Correct! :)
Glad you spotted that, forgot to mention when we were talking

Comment (francardoso93) [reply package.json:10]: And I agree about moving this to template

**Review by francardoso93 — APPROVED** [review]

**Review by simone-sanfratello — APPROVED** [review]

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,67 @@
+# Stage one - install
+FROM node:16-alpine as base
```
[the remark this answers is not in the document]
Comment (alanshaw) [reply Dockerfile:2]: This is basically a copy of https://github.com/ipfs-elastic-provider/ipfs-elastic-provider-indexer-lambda/blob/main/Dockerfile so I assume it does!

**Pull request merged**